crates/torii/core/src/sql.rs

```diff
@@ -124,26 +117,31 @@ impl Sql {
 
         let id = Argument::FieldElement(self.world_address);
 
-        self.query_queue.enqueue(
-            "UPDATE contracts SET last_pending_block_world_tx = ? WHERE id = ?",
-            vec![last_pending_block_world_tx, id],
-            QueryType::Other,
-        );
+        self.executor.send(QueryMessage {
+            statement: "UPDATE contracts SET last_pending_block_world_tx = ? WHERE id = ?"
+                .to_string(),
+            arguments: vec![last_pending_block_world_tx, id],
+            query_type: QueryType::Other,
+        })?;
```

Comment:
Having a function `new` like could make the code easier to change instead of plain initialization all over the file.

Also if `Other` is the one being mostly use, we may have a `QueryMessage::new_other(statement, arguments)`. And the other could be created by `::new` giving the enum.